packages/apollo-server-plugin-base/src/index.ts

```diff
@@ -20,6 +21,15 @@ export interface ApolloServerPlugin {
 }
 
 export interface GraphQLRequestListener<TContext = Record<string, any>> {
+  didResolveDocument?(
```

Comment:
I'm not sure I see why this hook is needed, because it seems we could just as well rely on `didResolveOperation`? If we do need it, I think the name `didResolveDocument` may be a bit confusing if we don't have `document` (the parsed document) yet at this point. Especially because we have a document cache, it seems fine to invoke it after we're guaranteed to have a `document` though. (I may be missing something, but I couldn't find where this is invoked from.)

Reply:
This gets the text in even if parsing fails.

Reply:
Ah, that makes sense. I like the name `didResolveDocument`, but since we already use `document` in the sense of parsed document that may be confusing. Not sure what the right solution is here. We could conceivably rename `document` to `documentAST`, although that would be breaking. It would also free up `document` to mean the unparsed document though (what is currently called `originalDocumentString`).

Reply:
Maybe `didResolveDocumentText` or `didResolveDocumentString`? I kind of like text even though it's not a term used elsewhere.

Reply:
Yep, `didResolveDocumentText` to be consistent with `documentText`.